app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/offline/OfflineDatabaseService.kt

```diff
@@ -248,12 +248,35 @@ class OfflineDatabaseService internal constructor(
                 firebaseUser.uid,
                 User.Type.FIREBASE,
                 firebaseUser.displayName,
-                firebaseUser.email
+                firebaseUser.email,
+                getUser(User.Type.FIREBASE, firebaseUser.uid)!!.isAdmin,
             )
         )
         return fromUser(userDao.load(firebaseUser.uid, User.Type.FIREBASE)!!)
     }
 
+    //TODO optimize this transaction
+    override suspend fun setAdminAccess(
+        firebaseUser: FirebaseUser,
+        adminAccess: Boolean
+    ): OfflineUser {
+        userDao.update(
+            RoomUser(
+                firebaseUser.uid,
+                User.Type.FIREBASE,
+                firebaseUser.displayName,
+                firebaseUser.email,
+                adminAccess,
+            )
+        )
+        return fromUser(userDao.load(firebaseUser.uid, User.Type.FIREBASE)!!)
```

Comment:
I suggest using the ?. operator and return the non-updated user if getUser() returns null in combination with logging that the operation failed

Reply:
for that isAdmin should be of type boolean?  ? ( second question mark is a literal question mark )

Reply:
No, I mean the result of `userDao.load` The point would be to avoid crashes because of `NullPointerErrors` which could occur with the usage of `!!`